cpu/cc26xx_cc13xx/periph/uart.c

```diff
@@ -44,50 +45,70 @@
  */
 static uart_isr_ctx_t ctx[UART_NUMOF];
 
-
 int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 {
     /* make sure the uart device is valid */
-    if (uart != 0) {
+    if (uart > UART_NUMOF) {
         return UART_NODEV;
     }
 
+#if defined(CPU_VARIANT_X2)
```

Comment:
I think a `#if defined(UART1_TX_PIN)` would be more generic.
The intent here is that only `CPU_VARIANT_X2` has UART1, right?

Reply:
```C
int tx_pin;
int rx_pin;
switch (uart) {
case 0:
    tx_pin = UART0_TX_PIN;
    …
    break;
#ifdef UART1_TX_PIN
case 1:
    tx_pin = UART1_TX_PIN;
    …
    break;
#endif
}
```

Mind you that other platforms just use a `uart_conf_t` struct and avoid having individual `#ifdefs` for each pin.